javalib/src/main/scala/java/math/BigInteger.scala

```diff
@@ -456,7 +456,8 @@ class BigInteger extends Number with Comparable[BigInteger] {
       val2
     } else if (val2.signum() == 0) {
       val1
-    } else if ((val1.numberLength == 1) && (val1.digits(0) > 0)) {
+    } else if (((val1.numberLength == 1) && (val1.digits(0) > 0))
+          && ((val2.numberLength == 1) && (val2.digits(0) > 0))) {
```

Comment:
`&&` should be at the end of the previous line. And this line should be indented 4 spaces wrt `}`.